src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -358,6 +358,19 @@ private function getAutowiredReference(TypedReference $reference): ?TypedReferen
             return $reference;
         }
 
+        if ($filterType && preg_match('/([&|])/', $type, $m)) {
+            $types = explode($m[0], $type);
+
+            if ('|' === $m[0]) {
+                // remove built-in types
+                $types = array_filter($types, function ($type) { return !\in_array(strtolower($type), ['int', 'string', 'array', 'bool', 'float', 'iterable', 'object', 'mixed', 'null'], true); });
```

Comment:
let's not care about the case and reduce this to, wdyt?
`$types = array_diff($types, [...]);`
we should remove `mixed` and add `callable` I think

Reply:
`array_diff`! That's what I was looking for :)